### PR TITLE
Add support for Vertex AI custom models deployed on private endpoint

### PR DESCRIPTION
**Pull request by RenaLu**
Add support for Vertex AI Model Garden models deployed on private endpoints. Example usage:
```
import litellm
import os

## set ENV variables
os.environ["VERTEXAI_PROJECT"] = "myproject"
os.environ["VERTEXAI_LOCATION"] = "us-central1"
os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = "credentials.json"

private_endpoint_id = "PRIVATE_ENDPOINT_ID"
response = litellm.completion(
  model=f"vertex_ai/private",
  model_id="private_endpoint_id"
  messages=[{ "content": "Hello, how are you?","role": "user"}],)
```

NOTE: streaming and async prediction in private mode doesn't work due to [google aiplatform issue](https://github.com/googleapis/python-aiplatform/issues/3307)

### CODE REVIEW

**Comment by ishaan-jaff** [comment]
Suggestion on the interface - feel free to ignore this 

We do this for Bedrock Provisioned throughput models (which are private endpoints) we just use the model_id param - which I think is a bit simpler than remembering to set `private=True` , this makes cost tracking / logging simpler since we can use the `model` for cost tracking 

```python
import litellm
response = litellm.completion(
    model="vertex_ai/llama2",
    model_id= private_endpoint_id,
    messages=[{"content": "Hello, how are you?", "role": "user"}]
)
```

**Comment by RenaLu** [comment]
> Suggestion on the interface - feel free to ignore this
> 
> We do this for Bedrock Provisioned throughput models (which are private endpoints) we just use the model_id param - which I think is a bit simpler than remembering to set `private=True` , this makes cost tracking / logging simpler since we can use the `model` for cost tracking
> 
> ```python
> import litellm
> response = litellm.completion(
>     model="vertex_ai/llama2",
>     model_id= private_endpoint_id,
>     messages=[{"content": "Hello, how are you?", "role": "user"}]
> )
> ```

Great suggestion!! Let me do that

**Comment by ghost** [comment]
Notes from call 
- does model garden handle chat templating 
- have private be part of the model name so you can use it within llama bot `model="vertex_ai/private"`

**Comment by ishaan-jaff** [comment]
## Remaining items 
- @RenaLu Can you upload a screenshot of the litellm.acompletion / acompletion calls working with the private endpoint ?

File: litellm/llms/vertex_ai.py [hunk 1]
```diff
@@ -532,9 +539,43 @@ def completion(
             """
             Vertex AI Model Garden
             """
+            ## LOGGING
```
Comment (ishaan-jaff) [inline litellm/llms/vertex_ai.py:542]: any reason for the changes to this branch ? I thought the pr was for `mode=private`

File: litellm/utils.py [hunk 2]
```diff
@@ -4192,7 +4192,14 @@ def _map_and_modify_arg(supported_params: dict, provider: str, model: str):
                 optional_params["stop_sequences"] = stop
         if max_tokens is not None:
             optional_params["max_output_tokens"] = max_tokens
-    elif custom_llm_provider == "vertex_ai":
+    elif custom_llm_provider == "vertex_ai" and model in (
+        litellm.vertex_chat_models
```
Comment (ishaan-jaff) [inline litellm/utils.py:4196]: can we not do this?  I can see a regression occuring for this very easily in the future 

lets stick to the convention of only checking `custom_llm_provider == "vertex_ai":`

**Review by ishaan-jaff — CHANGES_REQUESTED** [review]

File: litellm/utils.py (hunk 2 above)
Comment (ishaan-jaff) [reply litellm/utils.py:4196]: nvm saw discord convo

File: litellm/llms/vertex_ai.py (hunk 1 above)
Comment (RenaLu) [reply litellm/llms/vertex_ai.py:542]: I added info in models request string and updated it to make `llm_model` have a more similar usage to it's definitions with the other non-custom models

**Comment by RenaLu** [comment]
> ## Remaining items
> * @RenaLu Can you upload a screenshot of the litellm.acompletion / acompletion calls working with the private endpoint ?

acompletion for private endpoints does not work due to google's aiplatform support. I have linked the issue in my PR description and made a note of it in the code

**Review by ishaan-jaff — APPROVED** [review]

**Comment by ghost** [comment]
stabilizing main and then merging this PR.

**Pull request merged**